sdk/src/Services/DynamoDBv2/Custom/DataModel/ScanConfig.cs

```diff
@@ -0,0 +1,102 @@
+ */
+
+using Amazon.DynamoDBv2.DocumentModel;
+using System;
+using System.Collections.Generic;
+
+namespace Amazon.DynamoDBv2.DataModel
+{
+    /// <summary>
+    /// Input for the Scan operation in the object-persistence programming model
+    /// </summary>
+#if NET8_0_OR_GREATER
+    // The DataModel namespace doesn't support trimming yet, so annotate public classes/methods as incompatible
+    [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(Custom.Internal.InternalConstants.RequiresUnreferencedCodeMessage)]
+#endif
+    public class ScanConfig : BaseOperationConfig
+    {
+        /// <summary>
+        /// Indicates the name of the index to scan against.
+        /// This value is optional if the index name can be inferred from the call.
+        /// </summary>
+        public string IndexName { get; set; }
+
+        /// <summary>
+        /// The logical operator to apply to the filter conditions.
+        /// </summary>
+        /// <remarks>
+        /// <list type="bullet">
+        ///     <item>
+        ///         <term><see cref="ConditionalOperatorValues.And" /></term>
+        ///         <definition>If all of the conditions evaluate to true, then the entire filter evaluates to true.</definition>
+        ///     </item>
+        ///     <item>
+        ///         <term><see cref="ConditionalOperatorValues.Or" /></term>
+        ///         <definition>If at least one of the conditions evaluate to true, then the entire filter evaluates to true.</definition>
+        ///     </item>
+        /// </list>
+        /// The default value is <see cref="ConditionalOperatorValues.And" />.
+        /// </remarks>
+        public ConditionalOperatorValues ConditionalOperator { get; set; }
+
+        /// <summary>
+        /// Filter for the Scan operation. Evaluates the query results and returns only
+        /// the matching values. If you specify more than one condition, then by default all of the
+        /// conditions must evaluate to true. To match only some conditions, set <see cref="ConditionalOperator"/> to <see cref="ConditionalOperatorValues.Or" />.
+        /// </summary>
+        /// <remarks>
+        /// Note: Conditions must be against non-key properties.
+        /// </remarks>
+        public List<ScanCondition> QueryFilter { get; set; }
+
+        /// <summary>
+        /// Directs <see cref="DynamoDBContext" /> to ignore null values when 
+        /// converting an object to a DynamoDB item. If the property is false 
+        /// (or not set), null values will be interpreted as directives to 
+        /// delete the specific attributes on the DynamoDB item.
+        /// </summary>
+        public bool? IgnoreNullValues { get; set; }
+
+        /// <summary>
+        /// Property that directs <see cref="DynamoDBContext" /> to use consistent reads.
+        /// If property is not set, behavior defaults to non-consistent reads.
+        /// </summary>
+        public bool? ConsistentRead { get; set; }
```

Comment:
nit: Add a link to https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadConsistency.html to explain the behavior of setting this property.